### ISSUE TITLE
[Bug Report] After converting from bfp8 to float type, the 16th bit is 0.

### ISSUE THREAD

**Issue by dongjin-na** [tenstorrent/tt-metal#4198]
**Describe the bug**
When a random float tile is converted to bfp8 then converted to float again, 16th bit of every element in the tile is 0.
I wonder if it's okay. If not, the last bit of mantissa appears to have been truncated in the convert_u32_to_bfp8 or unpack_bfp8_tiles_into_float_vec function in bfloat8.hpp.

**To Reproduce**
1. Run this python script. "0" is printed out.
```
import torch
import tt_lib as ttl
import struct

torch.set_printoptions(precision=32)


def to_cpu(npu_tensor, shape, *, cpu_layout=ttl.tensor.Layout.ROW_MAJOR):
    if npu_tensor is None:
        return None
    cpu_tensor = npu_tensor.cpu().to(cpu_layout).unpad_from_tile(
        shape).to_torch()
    return cpu_tensor


def to_npu(cpu_tensor,
           device,
           *,
           npu_layout=ttl.tensor.Layout.TILE,
           npu_dtype=ttl.tensor.DataType.BFLOAT8_B,
           padding_value=float('nan')):
    if cpu_tensor is None:
        return None
    npu_tensor = ttl.tensor.Tensor(
        cpu_tensor,
        npu_dtype).pad_to_tile(padding_value).to(npu_layout).to(device)
    return npu_tensor


def binary(num):
    return ''.join('{:0>8b}'.format(c) for c in struct.pack('!f', num))


shape = [1, 1, 32, 32]
a = torch.randn(shape, dtype=torch.float32)
device = ttl.device.CreateDevice(0)
b = to_npu(a, device)
d = to_cpu(b, shape)
cnt = 0
for y in d[0, 0]:
    for yy in y:
        if binary(float(yy))[15] == '1':
            cnt += 1

print(cnt)

```

**Expected behavior**
There may be value loss during the conversion process, but it seems strange that the 16th bit is always 0.

**Screenshots**
Instead of screenshots, I will give an example of the conversion process.
expressing 16 floats (one row) as integer
bd955afb 3f8d87aa 401dba1a 3fd98627 bea981c9 3eac453f bfdc0b3e 3e0d43e9 be45319b 3faefed8 3e1a51e0 bf5d1432 bfc9501d bf846399 3f34a453 bee96450

A shared exponent of that row is 0x80.

First 
The sign + mantissa of the first four floats above (0xbd955afb 0x3f8d87aa 0x401dba1a 0x3fd98627) is
0x82, 0x23, 0x4f, 0x36.

If we remove 16 bits of the first 4 floats, it is as follows.
0xbd95, 0x3f8d, 0x401d, 0x3fd9

Combining the sign + shared exp + mantissa calculated above,
0xc002, 0x4023, 0x404f, 0x4036.

When converting bfp8 tile to float tile, it is converted as follows.
bd800000 3f8c0000 401e0000 3fd80000.

**Please complete the following environment information:**
- OS: Ubuntu 20.04
- Version of software (eg. commit)

**Comment by ayerofieiev-tt** [comment]
Please see these anwers.
I expect this ticket can be closed.

https://github.com/tenstorrent/tt-metal/issues/15914#issuecomment-2539881112

https://github.com/tenstorrent/tt-metal/issues/14090#issuecomment-2539812874

**Comment by dongjin-na** [comment]
Thanks for the confirmation. I will close this issue.
